### ISSUE TITLE
A more powerful get_or_create method

### ISSUE THREAD

**Issue by lsc20051426** [coleifer/peewee#571]
Currently the get_or_create is pretty weak.
Here is the example from django:

```
try:
    obj = Person.objects.get(first_name='John', last_name='Lennon')
except Person.DoesNotExist:
    obj = Person(first_name='John', last_name='Lennon', birthday=date(1940, 10, 9))
    obj.save()
```

This pattern gets quite unwieldy as the number of fields in a model goes up. The above example can be rewritten using get_or_create() like so:

```
obj, created = Person.objects.get_or_create(first_name='John', last_name='Lennon',
                  defaults={'birthday': date(1940, 10, 9)})
```

Thanks~

**Comment by coleifer** [comment]
I think I will pass on making changes to the implementation. The method has been deprecated for some time and there is documentation on implementing it yourself:

http://docs.peewee-orm.com/en/latest/peewee/querying.html#get-or-create

**Comment by coleifer** [comment]
These changes are now included in the new version 2.6.0. The function behaves just like it's Django equivalent.

**Comment by lsc20051426** [comment]
Cool!
